ngsPETSc/ts.py

```diff
@@ -0,0 +1,203 @@
+"""
+This module wraps the PETSc TS time-stepping routines.
+"""
+
+from petsc4py import PETSc
+from ngsolve import GridFunction, BilinearForm
+from ngsPETSc import VectorMapping, Matrix
+
+class TimeStepper:
+    """
+    This class wraps the PETSc TS time-stepping routines.
+    NOTE: Still a work in progress, need to add the connection with the
+          mass matrix for explicit time-stepping schemes and compute the
+          Jacobian using the variational form F.
+    """
+    def __init__(self, fes, info, G=None, F=None, residual=None,
+                 jacobian=None, solverParameters=None, optionsPrefix=None):
+        self.fes = fes
+        dofs = fes.ParallelDofs()
+        self.second_order = False
+        self.trial = info[0]
+        self.t = info[1]
+        self.F = F
+        self.G = G
+        if "ngs_jacobian_mat_type" not in solverParameters:
```

Comment:
if you do this, you need to use `solverParameters={}` not `solverParameters=None`